src/vma/dev/dm_mgr.cpp

```diff
@@ -217,17 +214,15 @@ bool dm_mgr::copy_data(struct mlx5_wqe_data_seg* seg, uint8_t* src, uint32_t len
 		goto dev_mem_oob;
 	}
 
-	/* Copy data into the On Device Memory buffer.
-	 * The copy is done using a volatile pointer (and not using memcpy of the whole buffer) to prevent any
-	 * optimizations which can lead to unaligned writes.
-	 * The first memcpy is to support architectures without unaligned access support - this should be optimized
-	 * out for architectures with unaligned access.
-	 */
-	while (offset < length_aligned_8) {
-		memcpy(&data_64, src + offset, sizeof(data_64));
-		*((volatile typeof(data_64) *)(m_p_mlx5_dm->start_va + m_head + offset)) = data_64;
-		offset += sizeof(data_64);
-	};
+	// Initialize memcopy attributes
+	memset(&memcpy_attr, 0, sizeof(memcpy_attr));
+	vma_ibv_init_memcpy_dm(memcpy_attr, src, m_head, length_aligned_8);
+
+	// Copy data into the On Device Memory buffer.
+	if (vma_ibv_memcpy_dm(m_p_ibv_dm, &memcpy_attr)) {
+		dm_logfunc("Failed to memcopy data into the memic buffer %m");
+		return false;
+	}
```

Comment:
It is not clear why verbs api is not used in original implementation. But if changes in the pull request more readable.